partner_external_maps/partner_external_maps.py

```diff
@@ -0,0 +1,197 @@
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MapWebsite(models.Model):
+    _name = 'map.website'
+    _description = 'Map Website'
+
+    name = fields.Char(string='Map Website Name', required=True)
+    address_url = fields.Char(
+        string='URL that uses the address',
+        help="In this URL, {ADDRESS} will be replaced by the address.")
+    lat_lon_url = fields.Char(
+        string='URL that uses latitude and longitude',
+        help="In this URL, {LATITUDE} and {LONGITUDE} will be replaced by "
+        "the latitude and longitude (requires the module 'base_geolocalize')")
+    route_address_url = fields.Char(
+        string='Route URL that uses the addresses',
+        help="In this URL, {START_ADDRESS} and {DEST_ADDRESS} will be "
+        "replaced by the start and destination addresses.")
+    route_lat_lon_url = fields.Char(
+        string='Route URL that uses latitude and longitude',
+        help="In this URL, {START_LATITUDE}, {START_LONGITUDE}, "
+        "{DEST_LATITUDE} and {DEST_LONGITUDE} will be replaced by the "
+        "latitude and longitude of the start and destination adresses "
+        "(requires the module 'base_geolocalize').")
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    # begin with context_ to allow user to change it by himself
+    context_map_website_id = fields.Many2one(
+        'map.website', string='Map Website',
+        domain=[
+            '|', ('address_url', '!=', False), ('lat_lon_url', '!=', False)])
+    # We want to give the possibility to the user to have one map provider for
+    # regular maps and another one for routing
+    context_route_map_website_id = fields.Many2one(
+        'map.website', string='Route Map Website',
+        domain=[
+            '|',
+            ('route_address_url', '!=', False),
+            ('route_lat_lon_url', '!=', False)],
+        help="Map provided used when you click on the car icon on the partner "
+        "form to display an itinerary.")
+    context_route_start_partner_id = fields.Many2one(
+        'res.partner', string='Start Address for Route Map')
+
+    @api.model
+    def _default_map_settings(self):
+        '''Method called from post-install script
```

Comment:
Although you need the post-init script for the existing users, it still would be useful to provide a default value for new created users since the installation of the module.